012-improve-entity-matching.md

```diff
@@ -23,55 +23,125 @@ the query results returned on import.
+- In `limit_scope` if classification is `committee` AND `committee_type` = `subcommittee`, then add the 
+`parent_committee_id` to query spec
+
+#### Solutions:
+- Core: Fix `limit_spec` on the `OrganizationImporter` so that more than just the first string in `other_names` is checked for
+Committees
+- People Script: Update Committee Script to include `other_names` for Committees that include Chamber, Type, & Both
 
+### Bill Matching to Event Agenda Items
 When it comes to matching Bills to Agenda Items on Events, I'm a little more fuzzy. Right now we have a [resolve_bill](https://github.com/openstates/openstates-core/blob/ac8e53aefe2a70d8ff360fc8b641bf77f28e2d7c/openstates/importers/base.py#L164)
 function on the `BaseImporter` that attempts to match Bills via `bill_id`, `jurisdiction_id`, & `date` if it gets passed,
 which seems like it could be improved by incorporating some of the logic in `resolve_related_bills` that Jesse worked on
 this spring where the match query is also narrowed down by `session_id`. We can certainly pass in more data to try to
 identify the Bill match better, but could also incorporate a LLM so will be testing out different approaches.
 
+#### Solutions:
+- Scrapers: Ensure `bill_identifier` matches the format of the expected Bill per jurisdiction
+- Core: Bill Identifier match improvements, passing in more data (at least `session`, maybe `chamber`)
+- Core: Add LLM to try better matching with above Core improvement
+- Core: Potentially cli command to try matching Events with Unmatched Bills in their agendas to Bills post-import
 
 ## Rationale
 
+### Bills or Votes to People or Committees
 We've known that matching Bills or Votes to Sponsors has been tricky for a while, hence OSEP #3 to help alleviate some
 of the issues with mismatching legislators. The People Matcher Tool can only get us so far, since we run into a blocker
 when there are legislators with the same last name in a jurisdiction or the sponsor is actually a committee, where
 adding an `other_name` to a person's yaml file isn't a possible fix.
 
+Current example for matching a Person to a Bill Sponsor:
+- Bill scraper calls `add_sponsorship` passing in { "name": "JOHNSON", entity_type="person", "classification"="primary",
+"primary"=True }
+- `add_sponsorship` creates a `pseudo_person_id` that is JOHNSON
+- BillImport calls `resolve_person` passing in that `pseudo_person_id` with start/end date values from the Bill's `session`
+- [resolve_person](https://github.com/openstates/openstates-core/blob/7ac7b73bbb0956f7a539128f9186929509c19550/openstates/importers/base.py#L526)
+constructs a spec that is used to compose filters to query data from the Person model to find a match. Could pass in
+`org_classification` but currently don't to narrow down via chamber
+- If jurisdiction has more than one legislator with the last name "Johnson", Importer will give an error message that
+`multiple people returned for spec` but continue through Import task
```

Comment:
I am imagining that "multiple people returned for spec" will be limited if you have organization classification in the resolve person query. 

Thinking of an idea here: 
If there are can we have these type of mismatch save somewhere in the database or a file that can be used in matching tool. I envision a matching tool that is broader than just the unmatched name and list of all the person. Have an idea of the multiple object returned can help quickly resolve the mismatch if and when they happen. This might be a good way to also gather data of mismatch. Some like 
```
unknown_entity | possible_matches | is_resolved |
JOHNSON.         | JOHNSON, Bill.      | False.          |
JOHNSON.         | JOHNSON, Mike.   | False.          |
```